code/GlobalNavControllerExtension.php

```diff
@@ -6,7 +6,7 @@ public function onBeforeInit() {
 		$host = GlobalNavSiteTreeExtension::get_toolbar_hostname();
 		if (
 				(isset($_REQUEST['regenerate_nav']) || isset($_REQUEST['flush'])) && 
-				$host == Director::absoluteBaseURL() && 
```

Comment:
The failure isn't in the PHP, it's in the template.

See GlobalNav.ss https://github.com/tractorcow/silverstripe-globaltoolbar/blob/pulls/2.0/fix-local-dev/templates/GlobalNavbar.ss#L155

The toolbarHostname would be assigned //localhost/basedir, which is the baseurl not the hostname.